src/st/prim_helpers.h

```diff
@@ -79,100 +79,102 @@ void UnkPrimHelper(Primitive* prim) {
 }
 
 s32 UpdateAnimation(u8* texAnimations, Primitive* prim) {
-    s16 sp0;
-    s16 tempUv;
-    u8 new_var;
-    u8* nextAnimation = texAnimations + ((prim->p1 * 5) & 0xFF);
-    u8 new_var2;
+    u16 sp0;
+    u16 tempUv;
     s32 retVal = 0;
+    u8 index = prim->p1 * 5;
+    u8* nextAnimation = &texAnimations[index];
 
-    if (prim->p2 == 0) {
+    if (!prim->p2) {
         if (*nextAnimation) {
             if (*nextAnimation == 0xFF) {
                 return 0;
             }
-            retVal = 0x80;
-            prim->p2 = nextAnimation[0];
-            ++nextAnimation;
+            prim->p2 = *nextAnimation++;
             tempUv = nextAnimation[0] + (nextAnimation[1] << 8);
             nextAnimation += 2;
             sp0 = nextAnimation[0] + (nextAnimation[1] << 8);
             LOH(prim->u0) = tempUv;
             LOH(prim->u1) = tempUv + *((u8*)(&sp0));
-            new_var = *((u8*)&sp0 + 1);
+            LOH(prim->u2) = tempUv + (*((u8*)&sp0 + 1) << 8);
             LOH(prim->u3) = tempUv + sp0;
-            LOH(prim->u2) = tempUv + (new_var << 8);
-            ++prim->p1;
+            prim->p1++;
+            retVal = (retVal | 0x80) & 0xFFFF;
         } else {
             prim->p1 = 0;
             prim->p2 = 0;
-            prim->p2 = texAnimations[0];
-            tempUv = texAnimations[1] + (texAnimations[2] << 8);
-            sp0 = texAnimations[3] + (texAnimations[4] << 8);
+            nextAnimation = &texAnimations[0];
+            prim->p2 = *nextAnimation++;
+            tempUv = nextAnimation[0] + (nextAnimation[1] << 8);
+            nextAnimation += 2;
+            sp0 = nextAnimation[0] + (nextAnimation[1] << 8);
             LOH(prim->u0) = tempUv;
             LOH(prim->u1) = tempUv + (*(u8*)&sp0);
-            new_var2 = *((u8*)&sp0 + 1);
+            LOH(prim->u2) = tempUv + (*((u8*)&sp0 + 1) << 8);
             LOH(prim->u3) = tempUv + sp0;
-            LOH(prim->u2) = tempUv + (new_var2 << 8);
-            ++prim->p1;
+            prim->p1++;
             return 0;
         }
     }
 
+    prim->p2--;
+#ifndef VERSION_PSP
     retVal |= 1;
```

Comment:
PSX is being finicky here, needs an extra ORI but I'm not sure why. If anyone else can work it out let me know

PSX: https://decomp.me/scratch/GDPlq
PSP: https://decomp.me/scratch/OEQHX